From 295621a873dfef4a8ed956fb0ad4563db0b2e659 Mon Sep 17 00:00:00 2001
From: Mike Westerhof <mwester@dls.net>
Date: Fri, 25 Jul 2008 23:06:18 +0100
Subject: [PATCH] commit 5f42e24d361cd83178fe8da9d68efbf41a011483

    Add missing initialization for the touchscreen driver for the
    gta01 platform.

    Signed-off-by: Mike Westerhof <mwester@dls.net>
---
 arch/arm/mach-s3c2410/mach-gta01.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-s3c2410/mach-gta01.c b/arch/arm/mach-s3c2410/mach-gta01.c
index a77ed3d..e518de7 100644
--- a/arch/arm/mach-s3c2410/mach-gta01.c
+++ b/arch/arm/mach-s3c2410/mach-gta01.c
@@ -494,8 +494,15 @@ static struct s3c2410_udc_mach_info gta01_udc_cfg = {
 
 static struct s3c2410_ts_mach_info gta01_ts_cfg = {
 	.delay = 10000,
-	.presc = 65,
+	.presc = 50000000 / 1000000, /* 50 MHz PCLK / 1MHz */
+	/* simple averaging, 2^n samples */
 	.oversampling_shift = 5,
+	/* averaging filter length, 2^n */
+	.excursion_filter_len_bits = 5,
+	/* flagged for beauty contest on next sample if differs from
+	 * average more than this
+	 */
+	.reject_threshold_vs_avg = 2,
 };
 
 /* SPI */
-- 
1.5.6.3

